### ISSUE TITLE
Deprecate renderer2, and and explainer on docs

### ISSUE THREAD

**Issue by SanderElias** [angular/angular#46940]
### Description

As `Renderer2` is mostly tied to Viewengine, it should be marked as deprecated.
Also, there should be a note in the docs explaining why renderer2 is no longer the recommended way to DOM manipulation.
This is needed because in the Angular community, there is still this idea that they should use it, and this idea is watched up by time, but most missed the notice.

### What is the affected URL?

nvt

### Please provide the steps to reproduce the issue

nvt

**Comment by pkozlowski-opensource** [comment]
I don't think we we are ready to depricate `Renderer2` or at least it requires more thinking / discussion.

**Comment by SanderElias** [comment]
@pkozlowski-opensource Even when it is not deprecated yet, it is still not the 'recommended' way to do DOM manipulation anymore. The docs still need a bit of an explainer around this.
Or, and that I also still possible, I'm completely wrong. If that is the case, the docs need some other explanation around this.

**Comment by ohois** [comment]
Out of curiosity what is the recommended way for DOM manipulation? I still have some instances where I'm using Renderer2

**Comment by SanderElias** [comment]
@ohois
```typescript
class SomeComponent {
   doc = inject(DOCUMENT)
   
   ngOninit() {
     const document = this.doc;te
     document.createElement('div')
     ... and so on
   }
}
```

So, inject `DOCUMENT` and use that like the old-fashioned way to do whatever you desire.

**Comment by bob-watson** [comment]
@pkozlowski-opensource do you have any suggestion where a good place to recommend this might be?

**Comment by bob-watson** [comment]
@jessicajaniuk, @pkozlowski-opensource, @SanderElias 
Reviving the question: where would be the best place to talk about this in the docs?

**Comment by SanderElias** [comment]
@bob-watson well, it depends 😜
seriously, if  renderer is going to be deprecated, it is a different kind of discussion. I would prefer to do that, as it makes it abundantly clear that this is no longer `the way™️`
If we keep it, we should document its updated role and why and for what it should be used.

**Comment by edusperoni** [comment]
Although I understand the reasoning behind it, the renderer interface might still be useful do use internally especially due to some extra optional APIs that can be used. One example is destroyNode, which is not in the dom spec but is useful for some clean ups on custom renderer implementations.

https://github.com/angular/angular/blob/6a88bad0192516f26a5a008c0634b73456b9447c/packages/core/src/render3/node_manipulation.ts#L95

Internally it seems to also be used for animations, where we need to intercept the node removal and sometimes delay the actual removal to a later time, which would be a bit troublesome to do if we had to override `DOCUMENT` with custom implementations of the methods needed:

https://github.com/angular/angular/blob/39b72e208b46d80f1d9a802cebf043c2ccf3c5f2/packages/platform-browser/animations/src/animation_renderer.ts#L22

https://github.com/angular/angular/blob/39b72e208b46d80f1d9a802cebf043c2ccf3c5f2/packages/platform-browser/animations/src/animation_renderer.ts#L176-L178

So maybe it having it moved to an internal API would make more sense? Then again, I'm not sure the implications would be if people are using Renderer2 specifically to trigger animations.

**Comment by kuncevic** [comment]
I always thought Renderer2 had been gone for a while now. Recently, while reviewing an Angular v17 app, I was surprised to discover that Renderer2 was being used. This revelation brought me here 😀

**Comment by JeanMeche** [comment]
Renderer2 is basically the base for building renderers on top of it. There is not plan to deprecate it today. We'll close this issue as not actionnable.

**Comment by chriswoodie** [comment]
@JeanMeche So... does this mean that `Renderer2` is not supposed to be used over simply injecting `DOCUMENT` and doing DOM manipulation that way if we need to add things like hreflang, ld+json schemas and whatever else?

Currently investigating a duplicated schema script being added when my app hydrates, and was wondering if using `Renderer2` (and therefore DOM manipulation) could be the culprit here, however the docs are a bit unclear on the topic.

**Comment by JeanMeche** [comment]
You don't have to use `Renderer2`, you can indeed reach for the DOM directly if you're on a regular app. 
`Renderer2` is an interface that abstracts the DOM apis for you. You might want to reach for it if you have a custom renderer (like what [AngularThree](https://angularthree.org/) uses for example).
